tmol/tests/score/constraint/test_constraint_energy_term.py

```diff
@@ -0,0 +1,187 @@
+import numpy
+import torch
+import math
+
+from tmol.score.constraint.constraint_energy_term import ConstraintEnergyTerm
+
+from tmol.tests.score.common.test_energy_term import EnergyTermTestBase
+from tmol.tests.score.common.test_energy_term import pose_stack_from_pdb_and_resnums
+from tmol.pose.pose_stack_builder import PoseStackBuilder
+
+from tmol.pose.constraint_set import ConstraintSet
+
+
+def add_test_constraints_to_pose_stack(pose_stack):
+    torch_device = pose_stack.device
+
+    constraints = ConstraintSet(device=torch_device)
+
+    # a distance constraint
+    cnstr_atoms = torch.full((1, 2, 3), 0, dtype=torch.int32, device=torch_device)
+    cnstr_params = torch.full((1, 1), 0, dtype=torch.float32, device=torch_device)
+
+    res1_type = pose_stack.block_type(0, 0)
+    res2_type = pose_stack.block_type(0, 1)
+    cnstr_atoms[0, 0] = torch.tensor([0, 0, res1_type.atom_to_idx["C"]])
+    cnstr_atoms[0, 1] = torch.tensor([0, 1, res2_type.atom_to_idx["N"]])
+    cnstr_params[0, 0] = 1.47
+
+    constraints.add_constraints(
+        ConstraintEnergyTerm.harmonic, cnstr_atoms, cnstr_params
+    )
+
+    # a bounded constraint
+
+    cnstr_atoms = torch.full((1, 2, 3), 0, dtype=torch.int32, device=torch_device)
+    cnstr_params = torch.full((1, 4), 0, dtype=torch.float32, device=torch_device)
+
+    res1_type = pose_stack.block_type(0, 0)
+    res2_type = pose_stack.block_type(0, 1)
+    cnstr_atoms[0, 0] = torch.tensor([0, 4, res1_type.atom_to_idx["C"]])
+    cnstr_atoms[0, 1] = torch.tensor([0, 5, res2_type.atom_to_idx["N"]])
+    cnstr_params[0, 0] = 1.47  # lb TODO
+    cnstr_params[0, 1] = 1.47  # ub TODO
+    cnstr_params[0, 2] = 1.47  # sd TODO
+    cnstr_params[0, 3] = 1.47  # rswitch TODO
```

Comment:
Some examples of a realistic bounded constraint would be very appreciated.

Reply:
something like 1,3,1,1